src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.cpp

```diff
@@ -87,21 +88,39 @@ std::set<std::vector<element::Type>> jit_brgemm_emitter::get_supported_precision
     }
 }
 
+cpu_isa_t jit_brgemm_emitter::get_primitive_isa(const ov::element::Type& dt_in0, bool is_with_amx) {
+    auto isa = isa_undef;
+#define SUPPORT(X, Y) if (mayiuse(X)) { isa = X; } else { Y }
+#define SUPPORT_ONE(X, MESSAGE) SUPPORT(X, OV_CPU_JIT_EMITTER_THROW(MESSAGE);)
+#define SUPPORT_TWO(X, Y, MESSAGE) SUPPORT(X, SUPPORT_ONE(Y, MESSAGE))
```

Comment:
Can we use variadic macros here to avoid `SUPPORT_ONE`, `SUPPORT_TWO`, `SUPPORT_THREE`? 😄

Reply:
> Variadic macro is good when we want to forward arguments to another function, but this is not the case.

Oh, right! Then the current version is ok for me 😃 
I really thought that there is recursion but It's not so